src/pages/context/cartcontext.js

```diff
@@ -0,0 +1,124 @@
+import { useContext , createContext } from "react";
+// eslint-disable-next-line no-unused-vars
+import { children } from "react";
+import { useReducer } from "react";
+import axios from 'axios'
+import { useState ,useEffect } from "react";
+const CartContext = createContext();
+
+const CartProvider = ({children}) =>{
+    const [originalproducts , setoriginalproducts] = useState([])
+    useEffect(()=>{
+        console.log('use effect running from cartcontext reducer func');
+        
+        axios.get('/api/products')
+        .then((response)=>{
+            setoriginalproducts(response.data.products)
+            dispatch({type:"setcart",payload:response.data.products})
+            
+          
+        },
+        (error)=>{
+          console.log(error);
+          
+        })
+    
+        
+    
+      },[])
```

Comment:
suggestion: use async await with IIFE

Reply:
Not very familiar with that syntax, so continuing with axios